FWCore/Reflection/interface/SetClassParsing.h

```diff
@@ -0,0 +1,24 @@
+#ifndef FWCore_Reflection_interface_SetClassParsing_h
+#define FWCore_Reflection_interface_SetClassParsing_h
+
+#include "TInterpreter.h"
+
+namespace edm {
+  /**
+   * An instance of this class can be used to temporarily enable or
+   * disable ROOT class parsing. The destructor restores the state.
+   *
+   * Because this class modifies the global state of ROOT, it should
```

Comment:
I'd like to see some detection or prevention of concurrent usage.  Could be just a static atomic bool and an assertion, to see if it ever happens.

Reply:
Ok, didn't work. And now this gets a bit hairy.

* The `cutParser()` needs auto-parsing enabled (as discussed earlier in this PR)
* The `cutParser()` is called from both module constructors (eager evaluation of the cut string) and from the module's event function (lazy evaluation of the cut string)
   * The `SetClassParsing` should change the auto parsing status only when used from the module construction
   * Strictly speaking `TCling::SetClassAutoParsing()` is not thread safe (the reads and writes are not atomic), so even writing the same value as the flag had from the event functions is technically an unsafe operation

I can think of two ways forward:
1. `cutParser()` knows if the evaluation is lazy or not. Initialize the guard only if the evaluation is not lazy.
2. Make the code region where the auto parsing is disable shorter, covering only the part where we check the dictionary existence

I'll give first a shot for the option 1 as it seems more straightforward, but I'm wondering if option 2 would be cleaner in the end?